### PR TITLE
Making homepage field url type to accept only url strictly

### PR DESCRIPTION
**Pull request by sudheesh001**
Fixes https://github.com/fedora-infra/anitya/issues/265

### CODE REVIEW

**Comment by pypingou** [comment]
Should we add some tests for this?

**Comment by sudheesh001** [comment]
@pypingou Could you guide me around that, I've seen the `test_project.py` file and most of it seems like the test for the data which is selected than the information which is to be inserted via the form. Since this is a front end change, does it really need the test ?

**Comment by pypingou** [comment]
You probably want to look more at: https://github.com/fedora-infra/anitya/blob/master/tests/test_flask.py#L249-L315

**Comment by sudheesh001** [comment]
@pypingou Is this what is required ?
```py
            falseData = {
                'name': 'repo_manager',
                'homepage': 'testurlrepo_manager',
                'backend': 'PyPI',
            }

            output = c.post(
                '/project/new', data=falseData, follow_redirects=True)
            self.assertTrue('Please enter a URL.' in output.data)
```
I am unable to run the tests because of the following error it throws up
```
>>> from tests import Modeltests, create_distro, create_project
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "tests/__init__.py", line 33, in <module>
    import vcr
ImportError: No module named vcr
```

**Comment by pypingou** [comment]
> ImportError: No module named vcr

Is listed in: https://github.com/fedora-infra/anitya/blob/master/test_requirements.txt

As for the tests, it looks good I think yes.

**Comment by sudheesh001** [comment]
@pypingou Thanks, i've made the changes and updated the PR after rebasing

File: tests/test_flask.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
 import unittest
 import sys
 import os
+import string
+import urlparse
```
Comment (pypingou) [inline tests/test_flask.py]: Why these two imports?

File: tests/test_flask.py [hunk 2]
```diff
@@ -275,6 +277,16 @@ def test_new_project(self):
...
 
+            falseData = {
+                'name': 'repo_manager',
+                'homepage': 'testurlrepo_manager',
+                'backend': 'PyPI',
+            }
+
+            output = c.post(
+                '/project/new', data=falseData, follow_redirects=True)
+            self.assertTrue('Please enter a URL.' in output.data)
```
Comment (pypingou) [inline tests/test_flask.py]: Maybe this should be moved up, we define data above ``falseData`` (which in python should be ``false_data``) but only use ``data`` below.

**Comment by pypingou** [comment]
Could you run the test suite?

**Comment by sudheesh001** [comment]
Yup i just ran the test suite after making the updates in the test and it throws the error as follows
```
======================================================================
FAIL: test_new_project (__main__.FlaskTest)
Test the new_project function.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_flask.py", line 286, in test_new_project
    self.assertTrue('Please enter a URL.' in output.data)
AssertionError: False is not true

----------------------------------------------------------------------
```

File: tests/test_flask.py [hunk 3]
```diff
@@ -276,6 +282,10 @@ def test_new_project(self):
             }
 
             output = c.post(
+                '/project/new', data=false_data, follow_redirects=True)
+            self.assertTrue('Please enter a URL.' in output.data)
+
```
Comment (pypingou) [inline tests/test_flask.py]: When speaking about moving up, I was thinking about the whole block not just ``false_data``.

**Comment by pypingou** [comment]
So it looks like your test has failed :)

**Comment by sudheesh001** [comment]
@pypingou I've made the changes, on second thought, It was easier to check for the strings on the success page and assertFalse() for this data as it shouldn't move to a success page because of the bad homepage URL. It passes the tests now.

**Comment by pypingou** [comment]
Anything I can help you with? :)

**Comment by ralphbean** [comment]
ping @sudheesh001 - can we help with resolving this?

**Comment by pypingou** [comment]
I opened https://github.com/fedora-infra/anitya/pull/307 based on the work from @sudheesh001 and w/ a little additional tweak :)

**Pull request closed without merging**